### PR TITLE
Milestone: game_logic

### PR DESCRIPTION
**Pull request by ajkacca457**
- Added logic to the game 

- checked whether the games work or not 

- Build different sections of the game separately and then assembled it together.

### CODE REVIEW

File: lib/board.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+  def display_board
+    puts " #{@board[0]} | #{@board[1]} | #{@board[2]} "
+    puts '-----------'
+    puts " #{@board[3]} | #{@board[4]} | #{@board[5]} "
+    puts '-----------'
+    puts " #{@board[6]} | #{@board[7]} | #{@board[8]} "
+    puts ' '
+  end
+end
```
Comment (kobiyoyo) [inline lib/board.rb]: - [x] No `puts/print` and `get.chomp` outside `main.rb` file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

File: lib/tictactoe.rb [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    puts "today our players are: #{@name[0].name} and #{@name[1].name}"
+    sleep 1
+    puts "#{@name[0].name} is going to play with 'X' symbol."
+    puts "#{@name[1].name} is going to play with 'O'symbol"
+    sleep 1
+    puts 'your game starts at:'
+    sleep 1
+    3.downto(0) { |number| puts number }
+    sleep 1
+  end
```
Comment (kobiyoyo) [inline lib/tictactoe.rb]: - [x] No `puts/print` and `get.chomp` outside `main.rb` file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

File: bin/main.rb [hunk 3]
```diff
@@ -1,2 +1,18 @@
...
+
+puts 'want to play again?'
+answer = gets.strip.downcase
+
+if answer == 'yes'
+  game = Tictactoe.new
+  game.execute
+else
+  puts 'Hope to see you soon'
+end
```
Comment (kobiyoyo) [inline bin/main.rb:129]: - [x] All game-logic related resources are kept in the `lib` directory. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone1/lib_dir_exist.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

File: lib/board.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# this is a class for setting the board for game
+class Board
+  attr_accessor :board
```
Comment (kobiyoyo) [inline lib/board.rb]: - [x] You should use  attr_readers or attr_writers. Only using attr_accessors it allows writing that property from outside the class. That variable can be set to anything. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/attr_accessor.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team     
#### Great Job so far, everything works fine, but there are a few more things required. 
### General
- [x] Commit messages should be written in an imperative form and not past tense.
- [x] You should add validations for empty input and integers.
- [x] Check comments below

### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

File: lib/board.rb (hunk 4 above)
Comment (ajkacca457) [reply lib/board.rb]: for the reviewers: unfortunately, this pull request got accidentally merged when we were merging the 2nd milestone. Based on the review we have added validation for empty inputs. We have put all the "puts" in the main file. We have created another pull request and used this pull request and reference in the comment.

**Pull request merged**